### DISCUSSION TITLE
Sample on patching a resource (ie Job)

### DISCUSSION THREAD

**Discussion by tomkerkhove** [kubernetes-client/csharp#1461] [Q&A] [answered]
I'm looking for ways to patch a resource based on previously known state (ie job) but keep getting `Method not allowed`.

Does anyone have a sample on how to achieve this?

This is not working for me:
https://github.com/kubernetes-client/csharp/blob/1ace761def84b55050ab094134480fcc0369195e/examples/customResource/Program.cs#L59-L66

This sample is purely based off of raw string which is not an option for me:
https://github.com/kubernetes-client/csharp/blob/1ace761def84b55050ab094134480fcc0369195e/examples/patch/Program.cs#L29

**Comment by tg123** [comment, marked as answer]
check generate json patch using library

https://github.com/kubernetes-client/csharp/blob/cdf5398e2d017f4687a15ba5f644aac47b469fc7/tests/E2E.Tests/MinikubeTests.cs#L129

all type inside `V1Patch.PatchType` are accepted, you have to put a jsonable object as param to V1Patch

**Reply by tomkerkhove** [reply]
This sample helps, but does not explain why I get a `Method not allowed` unfortunately.
